pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -226,6 +227,12 @@ func (r *SecretSubController) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, nil
 	}
 
+	r.managementCluster, err = utils.GetManagementCluster(ctx, r.client)
```

Comment:
I think it's a little bit fishy to store a per-reconcile object as a field on the reconciler to avoid passing it as an argument to a function. I'd rather we just pass this as an argument to `generateNamespacedSecrets`

Worried that storing it might lead to unexpected behaviors as this controller evolves.